work_queue/src/work_queue.c

```diff
@@ -851,31 +858,88 @@ static int get_transfer_wait_time(struct work_queue *q, struct work_queue_worker
 	return timeout;
 }
 
-static void remove_factory(struct work_queue_factory_info *f)
+static int factory_trim_workers(struct work_queue *q, struct work_queue_factory_info *f)
 {
-	free(f->name);
-	free(f);
+	char *factory_name;
+	if ( !f || !(factory_name = f->name) ) return 0;
```

Comment:
If f->name is null, that's a bug. Let's do:
```
if(!f) {
    return 0;
}

assert(f->name);
```
Below let's use `f->name` instead of `factory_name`. Less things to keep track of.